crates/circuit/src/dag_circuit.rs

```diff
@@ -4787,6 +4768,34 @@ impl DAGCircuit {
 }
 
 impl DAGCircuit {
+    pub fn new() -> PyResult<Self> {
+        Ok(DAGCircuit {
+            name: None,
+            metadata: None,
```

Comment:
Is there any effect from having the metadata be `None` instead of an empty dict during initialization? I'm assuming the effect should be minimal. But this would mean that any DAGCircuit coming fom rust directly would have no metadata.

Reply:
There shouldn't be because we allowed `None` as a type for the metadata. If we are returning a rust generated DAG to python it is already able to handle this: https://github.com/Qiskit/qiskit/blob/main/qiskit/converters/dag_to_circuit.py#L75 

The only reason I separated it out the python new though was because the default for Python's `DAGCircuit()` was intended to be an empty dict for metadata and I didn't want to change that expectation.